libcaf_openssl/src/middleman_actor.cpp

```diff
@@ -48,34 +50,53 @@ using native_socket = io::network::native_socket;
 using default_mpx = io::network::default_multiplexer;
 
 struct ssl_policy {
-  /// Reads up to `len` bytes from an OpenSSL socket.
-  static bool read_some(size_t& result, native_socket fd, void* buf,
+  ssl_policy(std::shared_ptr<ssl_session> ssl) : ssl_(ssl) {
```

Comment:
First, `ssl_(std::move(ssl))` safes two needless operations on the reference count. Second, we prefer `intrusive_ptr<...>` in CAF whenever feasible. All you need to do is to derive `session` from `ref_counted`.